### PR TITLE
clean up

### PR DESCRIPTION
**Pull request by awalterschulze**
tl;dr This is all just moving things around, cleanup, some extra comments and removing a few things.

Each heading describes a commit

## Cleanup Starlang

  - Moved somethings from Language.v to StarLang.v so that Language can be a little cleaner
  - Then added some more comments to StarLang.v
  - Removed some definitions and proofs that weren't really used in StarLang.v
  - Cleaned up the `destruct p as [p | a p].` statements to remove the warnings produced the coqc

## Create List Module

  - Moved theorems about lists in Decidable.v into CoqStock/List.v
  - Import everything with full qualified names, since List now has a name conflict

## Replace the Standard Library List Module

  - Use Require Export to export the List module and ListNotations from CoqStock.List, this way you only have to import CoqStock.List to get all the goodies.
  - Replace all imports outside of CoqStock itself of Coq.Lists.List and Coq.Lists.List.ListNotations with CoqStock.List

### CODE REVIEW

**Review by paulcadman — APPROVED** [review]
I like the use of `Require Export`.

Note (to self): We should document conventions like "use `CoqStock.List` instead of `Coq.Lists`".

**Comment by awalterschulze** [comment]
Sounds like a good idea.
Oh where would you like to document this?

**Comment by paulcadman** [comment]
I'm not sure - CONTRIBUTING.md seems like the right place.

**Comment by awalterschulze** [comment]
Ah yes, that can work.
A section on CoqStock can be useful.

**Comment by awalterschulze** [comment]
Although now that it is used everywhere, It might be easy enough to take the hint?

**Comment by paulcadman** [comment]
I think you're right
